source/Generic/PlayState/PlayState.cs

```diff
@@ -771,29 +773,32 @@ public override void OnGameStopped(OnGameStoppedEventArgs args)
                 }
             }
 
-            RemoveGameStopwatchTuple(game);
+            RemoveGame(game);
         }
 
-        private void CreateGameStopwatchTuple(Game game)
+        private void AddGame(Game game)
         {
-            if (stopwatchList.Any(x => x.Item1 == game.Id))
+            if (playStateData.Any(x => x.Game.Id == game.Id && x.Stopwatch != null))
             {
-                logger.Debug($"A stopwatch for game {game.Name} with id {game.Id} already exists");
+                logger.Debug($"Data for game {game.Name} with id {game.Id} already exists");
             }
             else
             {
-                stopwatchList.Add(new Tuple<Guid, Stopwatch>(game.Id, new Stopwatch()));
-                logger.Debug($"Stopwatch for game {game.Name} with id {game.Id} was created");
+                playStateData.Add(new PlayStateData(game, DateTime.Now, new Stopwatch()));
+                logger.Debug($"Data for game {game.Name} with id {game.Id} was created");
             }
         }
 
-        private void RemoveGameStopwatchTuple(Game game)
+        private void RemoveGame(Game game)
         {
-            var tuple = stopwatchList.FirstOrDefault(x => x.Item1 == game.Id);
-            if (tuple != null)
+            if (playStateData.Any(x => x.Game.Id == game.Id))
```

Comment:
Could use

```csharp
var gameData = FirstOrDefault(x => x.Game.Id == game.Id);
if (gameData != null)
{
    playStateData.Remove(gameData);
```

To prevent iterating twice